carbon-design-system/carbon
# fix(Pagination): disabled icon button tooltip & focus state for a11y

Closes #11032 

#### Changelog

- fixes an issue with the pagination next/prev icon buttons where the tooltip would stay visible even if the button would become disabled
- also fixes an a11y issue that arises by the fact that the disabled icon button loses focus and the `main` tag gets focused when this happens. This is a bad experience for keyboard and screen reader users. When discussed in CAG, the solution we came up with was that upon a prev/next button becoming disabled, the other button would gain focus. 

#### Testing / Reviewing

Go to pagination story
  - test next/previous buttons
    - click next all the way to the end until it becomes disabled
        1. tooltip should disappear immediately
        2. the previous btn should gain focus once the next becomes disabled
     - click the prev button all the way to the beginning
         1. tooltip should disappear immediately
         2. the next btn should gain focus when the prev btn becomes disabled
     - the same should happen when navigating via keyboard
